account_invoice_inter_company/models/account_move.py

```diff
@@ -48,18 +49,38 @@ def action_post(self):
             # do not consider invoices that have already been auto-generated,
             # nor the invoices that were already validated in the past
             dest_company = src_invoice._find_company_from_invoice_partner()
-            if not dest_company or src_invoice.auto_generated:
+            if not dest_company:
                 continue
             intercompany_user = dest_company.intercompany_invoice_user_id
             if intercompany_user:
                 src_invoice = src_invoice.with_user(intercompany_user).sudo()
             else:
                 src_invoice = src_invoice.sudo()
-            src_invoice.with_company(dest_company.id).with_context(
-                skip_check_amount_difference=True
-            )._inter_company_create_invoice(dest_company)
+            if not src_invoice.auto_generated:
+                src_invoice.with_company(dest_company.id).with_context(
+                    skip_check_amount_difference=True
+                )._inter_company_create_invoice(dest_company)
+            if src_invoice.is_sale_document():
+                src_invoice._attach_original_pdf_report()
         return res
 
+    def _attach_original_pdf_report(self):
+        supplier_invoice = self.auto_invoice_id
+        if not supplier_invoice:
+            supplier_invoice = self.search([("auto_invoice_id", "=", self.id)], limit=1)
+        report = self.env.ref("account.account_invoices").with_company(self.company_id)
+        pdf = report._render_qweb_pdf(report.report_name, [self.id])[0]
+        self.env["ir.attachment"].create(
+            {
+                "name": self.name + ".pdf",
```

Comment:
I'm thinking that the name may contain illegal characters for a file name, like `/`, so you should mask it. It may be also advisable to switch the method for `self` being the target invoice where the PDF is attached, as the method name suggest.

Reply:
and put `self.ensure_one()`.

Reply:
> I'm thinking that the name may contain illegal characters for a file name, like `/`, so you should mask it. 

Checking the process of creating an attachment through a report or sending an email, I confirm that no characters are replaced (although normally the name defined in the `_get_report_base_filename()` method is used so it has been changed).

> It may be also advisable to switch the method for `self` being the target invoice where the PDF is attached, as the method name suggest.

I think changing the whole method to do the opposite can be even more confusing, IMO it is clear, from a sales invoice, you create the attachment and link it to the purchase invoice.